compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingAttrs.cpp

```diff
@@ -65,26 +66,225 @@ static FailureOr<AffineMap> getComposedAffineMap(Attribute attr) {
+  FailureOr<ArrayAttr> elementTypes;
+  FailureOr<ArrayAttr> userIndexingMaps;
+  FailureOr<ArrayAttr> iterationSizes;
+  FailureOr<ArrayAttr> layouts;
+  if (failed(parseStruct(
+          p, AsmParser::Delimiter::LessGreater,
+          {"operand_index", "op_type", "element_types", "user_indexing_maps",
+           "iteration_sizes", "layouts"},
+          {[&]() {
+             operandIndex = mlir::FieldParser<IntegerAttr>::parse(p);
+             if (failed(operandIndex)) {
+               p.emitError(p.getCurrentLocation())
+                   << "failed to parse EncodingAttr parameter 'operand_index' "
+                      "which is to be a `IntegerAttr`";
+               return failure();
+             }
+             return success();
+           },
+           [&]() {
+             opType = FieldParser<EncodingOpTypeAttr>::parse(p);
+             if (failed(opType)) {
+               p.emitError(p.getCurrentLocation())
+                   << "failed to parse EncodingAttr parameter 'op_type' "
+                      "which is to be a `EncodingOpTypeAttr`";
+               return failure();
+             }
+             return success();
+           },
+           [&]() {
+             elementTypes = FieldParser<ArrayAttr>::parse(p);
+             if (failed(elementTypes)) {
+               p.emitError(p.getCurrentLocation())
+                   << "failed to parse EncodingAttr parameter 'element_types' "
+                      "which is to be a `ArrayAttr`";
+               return failure();
+             }
+             return success();
+           },
+           [&]() {
+             userIndexingMaps = FieldParser<ArrayAttr>::parse(p);
+             if (failed(userIndexingMaps)) {
+               p.emitError(p.getCurrentLocation())
+                   << "failed to parse EncodingAttr parameter "
+                      "'user_indexing_maps' which is to be a `ArrayAttr`";
+               return failure();
+             }
+             return success();
+           },
+           [&]() {
+             iterationSizes = parseDynamicI64ArrayAttr(p);
+             if (failed(iterationSizes)) {
+               p.emitError(p.getCurrentLocation())
+                   << "failed to parse EncodingAttr parameter "
+                      "'iteration_sizes' "
+                      "which is to be a `ArrayAttr`";
+               return failure();
+             }
+             return success();
+           },
+           [&]() {
+             layouts = FieldParser<ArrayAttr>::parse(p);
+             if (failed(layouts)) {
+               p.emitError(p.getCurrentLocation())
+                   << "failed to parse EncodingAttr parameter 'layouts' which "
+                      "is to be a `ArrayAttr`";
+               return failure();
+             }
+             return success();
+           }}))) {
+    p.emitError(p.getCurrentLocation()) << "failed parsing encoding attribute";
+    return {};
+  }
+  return p.getChecked<EncodingAttr>(
```

Comment:
Should we check if there is a failure in `operandIndex`, `opType`, and `elementTypes` as they are required arguments?

Reply:
Yes, good catch! Added those now.